sources/lib/util/options.ml

```diff
@@ -923,13 +966,13 @@ let parse_execution_opt =
     Arg.(value & opt string !vfrontend & info ["frontend"] ~docv ~docs ~doc) in
 
   let input_format =
-    let doc =
-      "Set the default input format to $(docv). Useful when the extension \
-       does not allow to automatically select a parser (eg. JS mode, GUI \
-       mode, ...)." in
+    let doc = Format.sprintf
+        "Set the default input format to $(docv) and must be %s. \
+         Useful when the extension does not allow to automatically select \
+         a parser (eg. JS mode, GUI mode, ...)."
+        (Arg.doc_alts ["native"; "smtlib"; "why3"]) in
     let docv = "FMT" in
-    Arg.(value & opt string !vinput_format &
-         info ["i"; "input"] ~docv ~doc) in
+    Arg.(value & opt (some string) None & info ["i"; "input"] ~docv ~doc) in
```

Comment:
It would be really really really really ..... really better to use cmdliner's converter here rather than string.

Reply:
Again, you should use cmdliner's convention to convert the argument into an `input_format`here instead of keeping a string and raison errors when you set the refs.